pkg/uploader/kopia/shim.go

```diff
@@ -0,0 +1,218 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kopia
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/content"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/repo/object"
+)
+
+type shimRepository struct {
+	udmRepo udmrepo.BackupRepo
+}
+
+type shimObjectWriter struct {
+	repoWriter udmrepo.ObjectWriter
+}
+
+type shimObjectReader struct {
+	repoReader udmrepo.ObjectReader
+}
+
+func NewShimRepo(repo udmrepo.BackupRepo) repo.RepositoryWriter {
+	return &shimRepository{
+		udmRepo: repo,
+	}
+}
+
+func (sr *shimRepository) OpenObject(ctx context.Context, id object.ID) (object.Reader, error) {
+	reader, err := sr.udmRepo.OpenObject(ctx, udmrepo.ID(id))
+	if reader == nil {
+		return nil, err
+	}
+
+	return &shimObjectReader{
+		repoReader: reader,
+	}, err
+}
+
+func (sr *shimRepository) VerifyObject(ctx context.Context, id object.ID) ([]content.ID, error) {
+	return nil, nil
+}
+
+func (sr *shimRepository) GetManifest(ctx context.Context, id manifest.ID, payload interface{}) (*manifest.EntryMetadata, error) {
+	repoMani := udmrepo.RepoManifest{
+		Payload: payload,
+	}
+
+	err := sr.udmRepo.GetManifest(ctx, udmrepo.ID(id), &repoMani)
+	return GetKopiaManifestEntry(repoMani.Metadata), err
+}
+
+func (sr *shimRepository) FindManifests(ctx context.Context, labels map[string]string) ([]*manifest.EntryMetadata, error) {
+	metadata, err := sr.udmRepo.FindManifests(ctx, udmrepo.ManifestFilter{
+		Labels: labels,
+	})
+
+	return GetKopiaManifestEntries(metadata), err
+}
+
+func GetKopiaManifestEntry(uMani *udmrepo.ManifestEntryMetadata) *manifest.EntryMetadata {
+	var ret manifest.EntryMetadata
+
+	ret.ID = manifest.ID(uMani.ID)
+	ret.Labels = uMani.Labels
+	ret.Length = int(uMani.Length)
+	ret.ModTime = uMani.ModTime
+
+	return &ret
+}
+
+func GetKopiaManifestEntries(uMani []*udmrepo.ManifestEntryMetadata) []*manifest.EntryMetadata {
+	var ret []*manifest.EntryMetadata
+
+	for _, entry := range uMani {
+		var e manifest.EntryMetadata
+		e.ID = manifest.ID(entry.ID)
+		e.Labels = entry.Labels
+		e.Length = int(entry.Length)
+		e.ModTime = entry.ModTime
+
+		ret = append(ret, &e)
+	}
+
+	return ret
+}
+
+func (sr *shimRepository) Time() time.Time {
+	return sr.udmRepo.Time()
+}
+
+func (sr *shimRepository) ClientOptions() repo.ClientOptions {
+	return repo.ClientOptions{}
+}
+
+func (sr *shimRepository) Refresh(ctx context.Context) error {
+	return nil
+}
+
+func (sr *shimRepository) ContentInfo(ctx context.Context, contentID content.ID) (content.Info, error) {
```

Comment:
Return something like "error not supported", the unified repo doesn't support this method

Reply:
Already modified it